packages/camera/camera_web/example/integration_test/camera_bitrate_test.dart

```diff
@@ -0,0 +1,146 @@
+        for (int x = 0; x < videoSize.width; x += w) {
+          canvasElement.context2D.setFillColorRgb(
+              random.nextInt(255), random.nextInt(255), random.nextInt(255));
+          canvasElement.context2D.fillRect(x, y, w, h);
+        }
+      }
+    }
+  }
+
+  setUpAll(() {
+    registerFallbackValue(MockCameraOptions());
+  });
+
+  testWidgets('Camera allows to control video bitrate',
+      (WidgetTester tester) async {
+    //const String supportedVideoType = 'video/webm';
+    const String supportedVideoType = 'video/webm;codecs="vp9,opus"';
+    bool isVideoTypeSupported(String type) => type == supportedVideoType;
+
+    Future<int> recordVideo(int videoBitrate) async {
+      final Window window = MockWindow();
+      final Navigator navigator = MockNavigator();
+      final MediaDevices mediaDevices = MockMediaDevices();
+
+      when(() => window.navigator).thenReturn(navigator);
+      when(() => navigator.mediaDevices).thenReturn(mediaDevices);
+
+      final CanvasElement canvasElement = CanvasElement(
+        width: videoSize.width.toInt(),
+        height: videoSize.height.toInt(),
+      )..context2D.fillRect(0, 0, videoSize.width, videoSize.height);
+
+      final VideoElement videoElement = VideoElement();
+
+      final MockCameraService cameraService = MockCameraService();
+
+      CameraPlatform.instance = CameraPlugin(
+        cameraService: cameraService,
+      )..window = window;
+
+      final CameraOptions options = CameraOptions(
+        audio: const AudioConstraints(),
+        video: VideoConstraints(
+          width: VideoSizeConstraint(
+            ideal: videoSize.width.toInt(),
+          ),
+          height: VideoSizeConstraint(
+            ideal: videoSize.height.toInt(),
+          ),
+        ),
+      );
+
+      final int cameraId = videoBitrate;
+
+      when(
+        () {
+          return cameraService.getMediaStreamForOptions(
+            options,
+            cameraId: cameraId,
+          );
+        },
+      ).thenAnswer(
+          (_) => Future<MediaStream>.value(canvasElement.captureStream()));
+
+      final Camera camera = Camera(
+          textureId: cameraId,
+          cameraService: cameraService,
+          options: options,
+          recorderOptions: (
+            audioBitrate: null,
+            videoBitrate: videoBitrate,
+          ))
+        ..isVideoTypeSupported = isVideoTypeSupported;
+
+      await camera.initialize();
+      await camera.play();
+
+      await camera.startVideoRecording();
+
+      await simulateCamera(canvasElement);
+
+      final XFile file = await camera.stopVideoRecording();
+
+      // Real movie can be saved locally during manual test invocation.
+      // First: add '--no-headless' to _targetDeviceFlags in
+      // `script/tool/lib/src/drive_examples_command.dart`, then uncomment:
+      // Second: uncomment next line
+      // await file.saveTo('movie.$videoBitrate.webm');
```

Comment:
Can confirm!

[377594011.webm](https://github.com/flutter/packages/assets/1255594/1e5e4ca7-3fb4-4417-8d98-362a97113cb9)

(Officially [nerd-sniped](https://jsfiddle.net/ditman/pzj82xr9/)!)

Reply:
but 377594011.webm link seems to be private...

Reply:
Weird, I uploaded it to github (seems to work in Chrome!)